pkg/operator/sync.go

```diff
@@ -738,6 +744,32 @@ func (optr *Operator) getGlobalConfig() (*configv1.Infrastructure, *configv1.Net
 	return infra, network, proxy, nil
 }
 
+func (optr *Operator) setEtcdOperatorImage(imgs *Images) error {
+	if optr.etcdLister == nil {
+		// if the resource is not found, i.e. it is not created by CVO
+		// which means cluster-etcd-operator images is not part of CVO
+		imgs.ControllerConfigImages.ClusterEtcdOperator = ""
+		glog.V(4).Info("etcd cr not found")
+		return nil
+	}
+	etcd, err := optr.etcdLister.Get("cluster")
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			imgs.ControllerConfigImages.ClusterEtcdOperator = ""
+			return nil
+		}
+		imgs.ControllerConfigImages.ClusterEtcdOperator = ""
+		return fmt.Errorf("error getting etcd CR: %#v", err)
+	}
+
+	if etcd.Spec.ManagementState == operatorv1.Unmanaged {
```

Comment:
If I'm following this right, the only case where we do want to set the etcd-operator image is when the `etcd.operator/cluster` has `spec.ManagementState == "Managed"`. Can we make that more clear by first gathering if its managed or not, similar to how we get `getGlobalConfig`, then set the controllerconfig values accordingly above?